### PR TITLE
Rebase of port/x64 branch

### PR DESCRIPTION
**Pull request by janisozaur**
I've rebased bits from old `port/x64` branch, updated them to work on current upstream and fixed some bits. This is only the first three commits so far, but I made them into something that can be merged, and I applied this to 64-bit job on Travis, so it doesn't brake.

### CODE REVIEW

**Comment by IntelOrca** [comment]
Looks fine, feel free to rebase it.

File: CMakeLists.txt [hunk 1]
```diff
@@ -44,6 +44,7 @@ option(DISABLE_NETWORK "Disable multiplayer functionality. Mainly for testing.")
 option(STATIC "Create a static build.")
 option(FORCE64 "Force native (x86-64) build. Do not use, for experimental purposes only.")
 option(DISABLE_OPENGL "Disable OpenGL support.")
+option(DISABLE_RCT2 "WIP: Try building without using things from vanilla.")
```
Comment (IntelOrca) [inline CMakeLists.txt]: code or instructions are better words than "things"

**Comment by IntelOrca** [comment]
Don't commit the change to the .vcxproj as it makes debug mode use `NO_RCT2`

**Comment by janisozaur** [comment]
I have updated this PR

**Pull request merged**